src/Controller/Api/FileController.php

```diff
@@ -51,7 +53,48 @@ public function deleteFile(File $file, MessageBusInterface $messageBus, EntityMa
 
         return new Response(
             null,
-            Response::HTTP_NO_CONTENT,
+            Response::HTTP_OK,
+            array(
+                'Content-Type' => 'application/x-empty',
+            )
+        );
+    }
+
+    /**
+     * Update a file entity.
+     *
+     * @param File                   $file          File entity instance.
+     * @param EntityManagerInterface $entityManager Entity manager interface instance.
+     * @param Request                $request       The request body sent with destination directory.
+     * @param Datastore              $datastore     Datastore to manipulate the file on disk.
+     *
+     * @Route("/api/file/{id}", name="pelagos_api_file_update_filename", methods={"PUT"}, defaults={"_format"="json"})
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_EDIT", subject="file")
+     *
+     * @throws BadRequestHttpException When the destination file name already exists.
+     *
+     * @return Response A response object with an empty body and a "no content" status code.
+     */
+    public function updateFileName(File $file, EntityManagerInterface $entityManager, Request $request, Datastore $datastore) : Response
+    {
+        $newFileName = $request->get('destinationDir');
+        $fileset = $file->getFileset();
+        if (!$fileset->doesFileExist($newFileName)) {
+            // Rename file on disk if it is already processed
+            if ($file->getStatus() === File::FILE_DONE) {
+                $datastore->renameFile($file->getFilePathName(), $newFileName);
```

Comment:
First parameter needs to be physicalFilePath, and renameFile needs to make a newFileName based on Datastore standards